### ISSUE TITLE
Add BigQuery support for arrayGet()

### ISSUE THREAD

**Issue by zineddine6** [jOOQ/jOOQ#17036]
### Expected behavior

PostgreSQL and BigQuery do not interpret this query `( my_array ) [ 1 ]` the same way.
Postgres retrieves the first element and BigQuery retrieves the second one (the count starts from 0).

The problem is that jOOQ translates the `DSL.arrayGet` method the same way for both DBs, which is problematic because the same jOOQ code will have different results depending on whether we are using Postgres or BQ. That's typically the case when Postgres is used locally and BQ when the app is deployed.

So the expected behavior would be to return always the same index of the array regardless of the DB.

A solution could be to use `ORDINAL` on BQ (1-based). For example, `( my_array ) [ ORDINAL(1) ]` in BQ will return the same thing as `( my_array ) [ 1 ]` in Postgres, ie. the first element of the array. 

### Actual behavior

`DSL.arrayGet(my_array, 1);` returns:
- The first element of 'my_array' in Postgres
- The second element in BQ

### Steps to reproduce the problem

This code:
```
public static final Field<String> COMPANY = field("company", SQLDataType.VARCHAR);
public static final Field<Integer> COMPANY_PRIORITY = field("company_priority", SQLDataType.INTEGER);
arrayGet(arrayAgg(COMPANY).orderBy(COMPANY_PRIORITY), inline(1));
```
will generate:
```
(
      array_agg(
        company
        order by
          company_priority
      )
) [ 1 ]
```
Which is not the same thing on PG and BQ.

Same issue with `arrayGet(arrayAgg(COMPANY).orderBy(COMPANY_PRIORITY), 1);`

### jOOQ Version

Entreprise edition 3.20-SNAPSHOT

### Database product and version

Latest BigQuery

### Java Version

openJDK 17.0.8.1 2023-08-24

**Comment by lukaseder** [comment]
Thank you very much for your report.

Please check the `@Support` annotation on the method `DSL.arrayGet()`. It does not currently list `BIGQUERY` as  a supported dialect:
https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/impl/DSL.html#arrayGet(org.jooq.Field,int)

In fact, most array functions aren't formally supported yet. I've created a feature request for this. Should be relatively quick to implement:

- https://github.com/jOOQ/jOOQ/issues/17037

This means we're not yet integration testing the methods yet on BigQuery, though, so I recommend not using them yet.

**Comment by lukaseder** [comment]
The main reason why we haven't looked into this topic yet is because the Simba JDBC driver doesn't support array bind variables, see the limitations here:

- https://github.com/jOOQ/jOOQ/issues/12058

But array bind variables aren't strictly necessary to support arrays in general, so I'll look at workarounds in our integration tests.

**Comment by lukaseder** [comment]
I've implemented this for the next 3.20.0-SNAPSHOT build. Will look into the other array functions tomorrow (#17037).

**Comment by lukaseder** [comment]
In fact, `SAFE_ORDINAL` should be used instead of `ORDINAL` to get the same behaviour as in most other dialects
